### PR TITLE
fix(core): custom resource providers cannot be used in CDK Pipelines

### LINKED ISSUE #11760: assets: EKS not deployable via CDK Pipelines
Hi there!

I am trying to deploy an EKS cluster via CDK Pipelines. This breaks, because the assets for custom resources within EKS are referenced with absolute path in `...assets.json`. This happens during `cdk synth` in the Pipelines build step. Then, in a later pipeline step, `cdk-assets` is not able to find the assets (because CodeBuild generate random temp dir names for each run) and therefor uploads empty zip files to S3. This breaks the deployment of the application stack (eks) with this error message in CloudFormation:

```
Uploaded file must be a non-empty zip (Service: AWSLambdaInternal; Status Code: 400; Error Code: InvalidParameterValueException; Request ID: 6443c97c-4806-4f82-8224-5a4e2afc48b2; Proxy: null)
```

### Reproduction Steps

1. Create a CDK Pipeline according to the [dev guide](https://docs.aws.amazon.com/cdk/latest/guide/cdk_pipeline.html)
1. add an EKS stack application stage via `pipeline.addApplicationStage()`
1. deploy the pipeline and wait for it to fail during application stack deployment

### What did you expect to happen?

`cdk synth` should create references with relative paths for the assets.

### What actually happened?

`cdk synth` creates references with absolute paths for the assets.

### Environment

  - **CDK CLI Version: 1.75** 
  - **Framework Version: 1.75**
  - **Node.js Version: v12.18.0** 
  - **OS               : Ubuntu 18.10**
  - **Language (Version): TypeScript (3.9.7)**

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by rix0rrr**
The Custom Resource Provider was using an absolute path staging assets.

Needed to be a relative path.

Fixes #11760.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by iliapolo** [comment]
@rix0rrr Seems like a test is in order..no?

**Comment by rix0rrr** [comment]
> @rix0rrr Seems like a test is in order..no?

I know :)

File: packages/@aws-cdk/core/lib/asset-staging.ts [hunk 1]
```diff
@@ -93,6 +93,9 @@ export class AssetStaging extends CoreConstruct {
    * a temporary directory used for bundling.
    *
    * If asset staging is enabled it will be the staged path.
+   *
+   * IMPORTANT: If you are going to call `addFileAsset()`, use
+   * `relativeStagedPath()` instead.
    */
   public readonly stagedPath: string;
```
Comment (eladb) [inline packages/@aws-cdk/core/lib/asset-staging.ts:97]: Should we even expose this? Maybe deprecate and call this `absoluteStagedPath`

**Review by eladb — APPROVED** [review]

File: packages/@aws-cdk/core/lib/asset-staging.ts (hunk 1 above)
Comment (rix0rrr) [reply packages/@aws-cdk/core/lib/asset-staging.ts:97]: Sure.

**Pull request merged**